pkgs/development/tools/misc/binutils/windres-locate-gcc.patch

```diff
@@ -0,0 +1,19 @@
+diff --git a/binutils/resrc.c b/binutils/resrc.c
+index a875c3a4..0411d047 100644
+--- a/binutils/resrc.c
++++ b/binutils/resrc.c
+@@ -521,7 +521,13 @@ read_rc_file (const char *filename, const char *preprocessor,
+ 
+       cpp_pipe = 0;
+ 
+-      if (dash)
++      /* Nixpkgs specific : look first at the prefixed path
++        ( there should be no gcc in the binutils folder ) */
++      if (slash && dash) {
++        cpp_pipe = look_for_default(cmd, slash + 1, dash - slash, preprocargs, filename);
++      }
++
++      if (dash && ! cpp_pipe)
+ 	{
+ 	  /* First, try looking for a prefixed gcc in the windres
+ 	     directory, with the same prefix as windres */
```

Comment:
What happens without this?

Reply:
The original code for this part was

```C
	if (dash)
	{
	  /* First, try looking for a prefixed gcc in the windres
	     directory, with the same prefix as windres */

	  cpp_pipe = look_for_default (cmd, program_name, dash - program_name + 1,
				       preprocargs, filename);
	}

      if (slash && ! cpp_pipe)
	{
	  /* Next, try looking for a gcc in the same directory as
             that windres */

	  cpp_pipe = look_for_default (cmd, program_name, slash - program_name + 1,
				       preprocargs, filename);
	}

      if (! cpp_pipe)
	{
	  /* Sigh, try the default */

	  cpp_pipe = look_for_default (cmd, "", 0, preprocargs, filename);
	}
```


The cpp_pipe is pointer to an open file (I think it's the gcc binary).


slash contain a pointer to the last ``/`` in the filename in memory, and dash a pointer to the last ``-``.

Then, it look for gcc (without this patch) in 3 different place (ordered by search order):

(where COMMAND is the command entered in the terminal)

if there is a dash, <COMMAND up to the last dash>
if not found and slash, <COMMAND up to the last slash>
default: <nothing> (seemingly equivalent to look for unprefixed gcc from PATH)

I added a fourth possibility, that will be tried first :
if there is a slash and a dash, then the binary was probably called with it's full name.
we cut the between the last slash and the last dash. That does something like (for COMMAND)
``a/b/c/something-with-dash`` will try with ``something-with-``.

Reply:
The logic you describe seems reasonable, even outside nixpkgs, maybe send the patch upstream ?

Reply:
I will do this eventually. I'm just a bit intimitided by the absence of gitlab/whatever for PR (guess I should send this on the bug mailing list). And the fact that ``nix develop nixpkgs#binutils`` doesn't work.